### PR TITLE
[AWS] Fill-in AMI if not provided

### LINKED ISSUE #13800: [Autoscaler] AWS autoscaler not automatically filling in ImageId
### What is the problem?

If no ImageId is specified, we used to have a default image, but now it errors.

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

Example yaml: https://github.com/ray-project/ray/blob/master/release/long_running_distributed_tests/cluster.yaml

```
Traceback (most recent call last):
  File "/home/ubuntu/.local/bin/ray", line 11, in <module>
    load_entry_point('ray', 'console_scripts', 'ray')()
  File "/home/ubuntu/ray/python/ray/scripts/scripts.py", line 1519, in main
    return cli()
  File "/home/ubuntu/.local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ubuntu/.local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/ray/python/ray/scripts/scripts.py", line 1171, in submit
    create_or_update_cluster(
  File "/home/ubuntu/ray/python/ray/autoscaler/_private/commands.py", line 226, in create_or_update_cluster
    try_logging_config(config)
  File "/home/ubuntu/ray/python/ray/autoscaler/_private/commands.py", line 73, in try_logging_config
    log_to_cli(config)
  File "/home/ubuntu/ray/python/ray/autoscaler/_private/aws/config.py", line 179, in log_to_cli
    print_info(
  File "/home/ubuntu/ray/python/ray/autoscaler/_private/aws/config.py", line 129, in print_info
    head_value_str = config["head_node"][key]
KeyError: 'ImageId'
```

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* If no AMI is given, this uses a recent DL AMI

## Related issue number

Closes #13800

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by wuisawesome — APPROVED** [review]

**Comment by AmeerHajAli** [comment]
Can we please a small test the verifies this?

File: python/ray/tests/aws/test_autoscaler_aws.py [hunk 1]
```diff
@@ -133,6 +134,17 @@ def test_subnet_given_head_and_worker_sg(iam_client_stub, ec2_client_stub):
...
 
 
+def test_fills_out_amis():
+    config = helpers.load_aws_example_config_file("example-full.yaml")
+    del config["head_node"]["ImageId"]
+    del config["worker_nodes"]["ImageId"]
+
+    defaults_filled = bootstrap_aws(config)
+
+    assert defaults_filled["head_node"].get("ImageId")
```
Comment (AmeerHajAli) [inline python/ray/tests/aws/test_autoscaler_aws.py:144]:
```suggestion
    assert defaults_filled["head_node"].get("ImageId") == "latest_dlami"
```

File: python/ray/tests/aws/test_autoscaler_aws.py [hunk 2]
```diff
@@ -133,6 +134,17 @@ def test_subnet_given_head_and_worker_sg(iam_client_stub, ec2_client_stub):
...
+    assert defaults_filled["worker_nodes"].get("ImageId")
```
Comment (AmeerHajAli) [inline python/ray/tests/aws/test_autoscaler_aws.py:145]:
```suggestion
    assert defaults_filled["head_node"].get("ImageId") == "latest_dlami"
```

**Review by AmeerHajAli — APPROVED** [review]
LGTM.
I committed two minors changes to assert it is "latest_dlami".

Thanks for the PR!

**Comment by AmeerHajAli** [comment]
It seems like the tests are still failing for credentials?
?

**Comment by ijrsvt** [comment]
> It seems like the tests are still failing for credentials?
> ?

Ah, I do my development on an AWS box, so I didn't realize that credential mocking was needed :) Fixed now!

**Comment by AmeerHajAli** [comment]
Can you run lint and the test locally before you push? It seems like the test and lint are failing. For the test, run it in a new terminal without the credentials? Or turn off your network/Wifi maybe?

**Comment by ijrsvt** [comment]
> Can you run lint and the test locally before you push? It seems like the test and lint are failing. For the test, run it in a new terminal without the credentials? Or turn off your network/Wifi maybe?

@AmeerHajAli I'm trying to figure out how to do this. I do not have a local development environment at the moment because I use ChromeOS.

**Comment by AmeerHajAli** [comment]
LGTM.
@ericl , can you please merge?

**Pull request merged**